### ISSUE TITLE
If one argument in an argument list is multiline, surrounding arguments should have their own lines

### ISSUE THREAD

**Issue by nex3** [dart-lang/dart_style#8]
Currently,

```dart
_log(new LogEntry(error.transform, error.transform.primaryId,
        LogLevel.ERROR, message, null),
    arg);
```

is formatted as

```dart
_log(new LogEntry(error.transform, error.transform.primaryId, LogLevel.ERROR,
    message, null), arg);
```

The second argument to `_log` is hidden after the long argument list for `new LogEntry()`. If one argument is multi-line, no part of it should be on the same line as other arguments.

**Comment by munificent** [comment]
Look at `includeSuperAndInjectedMembers` in before:

```dart
    classElement.forEachInstanceField(
        (ClassElement enclosingClass, VariableElement member) {
          HInstruction value = fieldValues[member];
          if (value == null) {
            // Uninitialized native fields are pre-initialized by the native
            // implementation.
            assert(isNativeUpgradeFactory);
          } else {
            fields.add(member);
            DartType type = localsHandler.substInContext(member.type);
            constructorArguments.add(potentiallyCheckOrTrustType(value, type));
          }
        },
        includeSuperAndInjectedMembers: true);
```

And after:

```dart
    classElement.forEachInstanceField(
        (ClassElement enclosingClass, VariableElement member) {
          HInstruction value = fieldValues[member];
          if (value == null) {
            // Uninitialized native fields are pre-initialized by the native
            // implementation.
            assert(isNativeUpgradeFactory);
          } else {
            fields.add(member);
            DartType type = localsHandler.substInContext(member.type);
            constructorArguments.add(potentiallyCheckOrTrustType(value, type));
          }
        },
        includeSuperAndInjectedMembers: true);
      HInstruction value = fieldValues[member];
      if (value == null) {
        // Uninitialized native fields are pre-initialized by the native
        // implementation.
        assert(isNativeUpgradeFactory);
      } else {
        fields.add(member);
        DartType type = localsHandler.substInContext(member.type);
        constructorArguments.add(potentiallyCheckOrTrustType(value, type));
      }
    }, includeSuperAndInjectedMembers: true);
```

**Comment by munificent** [comment]
And:

```dart
      HLoopBlockInformation info =
          new HLoopBlockInformation(
              HLoopBlockInformation.loopType(loop),
              wrapExpressionGraph(initializerGraph),
              wrapExpressionGraph(conditionExpression),
              wrapStatementGraph(bodyGraph),
              wrapExpressionGraph(updateGraph),
              conditionBlock.loopInformation.target,
              conditionBlock.loopInformation.labels,
              sourceFileLocationForBeginToken(loop),
              sourceFileLocationForEndToken(loop));
```

After:

```dart
      HLoopBlockInformation info = new HLoopBlockInformation(
          HLoopBlockInformation.loopType(loop),
          wrapExpressionGraph(initializerGraph),
          wrapExpressionGraph(conditionExpression),
          wrapStatementGraph(bodyGraph), wrapExpressionGraph(updateGraph),
          conditionBlock.loopInformation.target,
          conditionBlock.loopInformation.labels,
          sourceFileLocationForBeginToken(loop),
          sourceFileLocationForEndToken(loop));
```

**Comment by munificent** [comment]
Before:

```dart
      handleSwitch(node,
                   new NullJumpHandler(compiler),
                   buildExpression, node.cases, getConstants,
                   (_) => false, // No case is default.
                   buildSwitchCase);
    }
```

After:

```dart
      handleSwitch(node, new NullJumpHandler(compiler), buildExpression,
          node.cases, getConstants, (_) => false, // No case is default.
          buildSwitchCase);
    }
```

**Comment by nex3** [comment]
@munificent I'm not sure I see what you're getting at

**Comment by munificent** [comment]
Sorry, just adding more regression examples of the bug. :)

**Comment by munificent** [comment]
Fixed by 346ad0736b5aff21ce5751c818bd4ccfc02d5885!
